src/plugins/vis_augmenter/public/view_events_flyout/components/utils.tsx

```diff
@@ -0,0 +1,243 @@
+  savedObjectId: string,
+  setTimeRange: Function,
+  setVisEmbeddable: Function,
+  setErrorMessage: Function
+): Promise<void> {
+  try {
+    const embeddable = await fetchVisEmbeddable(savedObjectId);
+    setTimeRange(getQueryService().timefilter.timefilter.getTime());
+    setVisEmbeddable(embeddable);
+  } catch (err: any) {
+    setErrorMessage(String(err));
+  }
+}
+
+/**
+ * For each VisLayer in the base vis embeddable, generate a new filtered vis
+ * embeddable (based off of the base vis), and pass in extra arguments to only
+ * show datapoints for that particular VisLayer. Partition them by
+ * plugin resource type via an EventVisEmbeddablesMap.
+ * @param savedObjectId the saved object id of the base vis embeddable
+ * @param embeddable the base vis embeddable
+ * @param setEventVisEmbeddablesMap custom hook used in base component
+ * @param setErrorMessage custom hook used in base component
+ */
+export async function createEventEmbeddables(
+  savedObjectId: string,
+  embeddable: VisualizeEmbeddable,
+  setEventVisEmbeddablesMap: Function,
+  setErrorMessage: Function
+) {
+  const embeddableVisFactory = getEmbeddable().getEmbeddableFactory('visualization');
+  try {
+    const { left, right } = getValueAxisPositions(embeddable);
+    const map = new Map<string, EventVisEmbeddableItem[]>() as EventVisEmbeddablesMap;
+    // Currently only support PointInTimeEventVisLayers. Different layer types
+    // may require different logic in here
+    const visLayers = (get(embeddable, 'visLayers', []) as VisLayer[]).filter((visLayer) =>
+      isPointInTimeEventsVisLayer(visLayer)
+    ) as PointInTimeEventsVisLayer[];
+    if (visLayers !== undefined) {
+      const contextInput = {
+        filters: embeddable.getInput().filters,
+        query: embeddable.getInput().query,
+        timeRange: embeddable.getInput().timeRange,
+      };
+
+      await Promise.all(
+        visLayers.map(async (visLayer) => {
+          const pluginResourceType = visLayer.pluginResource.type;
+          const eventEmbeddable = (await embeddableVisFactory?.createFromSavedObject(
+            savedObjectId,
+            {
+              ...contextInput,
+              visAugmenterConfig: {
+                visLayerResourceIds: [visLayer.pluginResource.id as string],
+                inFlyout: true,
+                flyoutContext: VisFlyoutContext.EVENT_VIS,
+                leftValueAxisPadding: left,
+                rightValueAxisPadding: right,
+              },
+            } as VisualizeInput
+          )) as VisualizeEmbeddable | ErrorEmbeddable;
+
+          if (eventEmbeddable instanceof ErrorEmbeddable) {
+            throw getErrorMessage(eventEmbeddable);
+          }
+
+          eventEmbeddable.updateInput({
+            // @ts-ignore
+            refreshConfig: {
+              value: 0,
+              pause: true,
+            },
+          });
+
+          const curList = (map.get(pluginResourceType) === undefined
```

Comment:
Do we care for it to be specifically `undefined`? if not, `.has()` would be a better choice.